native/wasm/build.cake

```diff
@@ -0,0 +1,91 @@
+        $"is_official_build=true " +
+        $"is_static_skiasharp=true " +
+        $"skia_enable_ccpr=false " +
+        $"skia_enable_fontmgr_custom_empty=false " +
+        $"skia_enable_fontmgr_empty=false " +
+        $"skia_enable_gpu={(SUPPORT_GPU ? "true" : "false")} " +
+        (SUPPORT_GPU ? "skia_gl_standard='webgl'" : "") +
+        $"skia_enable_nvpr=false " +
+        $"skia_enable_pdf=false " +
+        $"skia_enable_tools=false " +
+        $"skia_use_dng_sdk=false " +
+        $"skia_use_egl=true " +
+        $"skia_use_fontconfig=false " +
+        $"skia_use_freetype=true " +
+        $"skia_use_harfbuzz=false " +
+        $"skia_use_icu=false " +
+        $"skia_use_piex=false " +
+        $"skia_use_sfntly=false " +
+        $"skia_use_system_expat=false " +
+        $"skia_use_system_freetype2=false " +
+        $"skia_use_system_libjpeg_turbo=false " +
+        $"skia_use_system_libpng=false " +
+        $"skia_use_system_libwebp=false " +
+        $"skia_use_system_zlib=false " +
+        $"skia_use_vulkan=false " +
+        $"skia_use_wuffs=true " +
+        $"use_PIC=false " +
+        $"werror=true " +
+        $"extra_cflags=[ " +
+        $"  '-DSKIA_C_DLL', '-DXML_POOR_ENTROPY', " + 
+        $"  '-DSK_DISABLE_READBUFFER', '-DSK_DISABLE_EFFECT_DESERIALIZATION', " +
+        $"  '-s', 'WARN_UNALIGNED=1', '-DSKNX_NO_SIMD', '-DSK_DISABLE_AAA', '-DGR_GL_CHECK_ALLOC_WITH_GET_ERROR=0' ] " +
+        $"extra_cflags_cc=[ '-frtti' ] " +
+        compilers +
+        ADDITIONAL_GN_ARGS);
+
+    var a = SKIA_PATH.CombineWithFilePath($"out/wasm/libSkiaSharp.a");
+
+    // separate all the .a files into .o files
+    var skiaOut = SKIA_PATH.Combine("out/wasm");
+    var mergeDir = skiaOut.Combine("obj/merge");
+    EnsureDirectoryExists(mergeDir);
+    CleanDirectories(mergeDir.FullPath);
+    foreach (var file in GetFiles($"{skiaOut}/*.a")) {
+        RunProcess("ar", new ProcessSettings {
+            Arguments = $"x \"{file}\"",
+            WorkingDirectory = mergeDir.FullPath,
+        });
+    }
+
+    // add the default font
+    var input = SKIA_PATH.CombineWithFilePath("modules/canvaskit/fonts/NotoMono-Regular.ttf");
+    var embed_resources = SKIA_PATH.CombineWithFilePath("tools/embed_resources.py");
+    RunProcess(PYTHON_EXE, new ProcessSettings {
+        Arguments = $"{embed_resources} --name SK_EMBEDDED_FONTS --input {input} --output {input}.cpp --align 4",
+        WorkingDirectory = SKIA_PATH.FullPath,
+    });
+    RunProcess("emcc", $"-std=c++17 -I. {input}.cpp -r -o {mergeDir}/NotoMonoRegularttf.o");
+
+    // merge all the .o files into the final .a file
+    var oFiles = GetFiles($"{mergeDir}/*.o");
+    RunProcess("ar", $"-crs {a} {string.Join(" ", oFiles)}");
+
+    var outDir = OUTPUT_PATH.Combine($"wasm");
+    EnsureDirectoryExists(outDir);
+    CopyFileToDirectory(a, outDir);
+});
+
+Task("Default")
+    .IsDependentOn("libSkiaSharp");
+    // TODO: .IsDependentOn("libHarfBuzzSharp");
```

Comment:
https://github.com/harfbuzz/harfbuzzjs

Reply:
I did a bit of looking around... seems emscripten already comes with harfbuzz... We might just be able to pinvoke into that directly 🤔  Need to research how to do that, but then it just has a managed part... #DoTheResearch